flink-runtime/src/test/java/org/apache/flink/runtime/source/coordinator/SourceCoordinatorAlignmentTest.java

```diff
@@ -131,6 +133,34 @@ void testWatermarkAlignmentWithTwoGroups() throws Exception {
         }
     }
 
+    @Test
+    @Timeout(value = 10)
```

Comment:
Also as a general rule, we are never adding timeouts on per test basis:

- we have some watchdog implemented in the CI, that after some inactivity OR global ~4h timeout, timeouts all running tests with thread dump and some other extra things
- timeouts such as this, tend to fail in the CI infrastructure, as sooner or later some crappy machine running this particular test will freeze for whatever reason , even for minutes.
- it is is annoying when such test fails during debugging from IDE

Reply:
got it, will update later~